src/corehost/src/main.cpp

```diff
@@ -43,19 +66,20 @@ int run(arguments_t args, pal::string_t app_base, tpafile tpa)
     else
     {
         trace::info(_X("using packages directory: %s"), packages_dir.c_str());
-        tpa.add_package_dir(packages_dir);
     }
 
-    // Add native search path
-    trace::info(_X("using native search path: %s"), packages_dir.c_str());
-    tpa.add_native_search_path(args.clr_path);
-
     // Build TPA list and search paths
     pal::string_t tpalist;
-    tpa.write_tpa_list(tpalist);
+    tpa->write_tpa_list(args.app_dir, packages_dir, clr_path, tpalist);
+
+    trace::info(_X("using native search path: %s"), packages_dir.c_str());
 
     pal::string_t search_paths;
-    tpa.write_native_paths(search_paths);
+    tpa->write_native_paths(args.app_dir, packages_dir, clr_path, search_paths);
+
+    // TODO:
+    // pal::string_t culture_paths;
```

Comment:
We should implement this. Since there will not be any entries, the operation will be a no-op for now and will just lightup when the entries come in.